### PR TITLE
CLN: Clean reductions/test_reductions.py

### PR DESCRIPTION
**Pull request by dsaxton**
Some more parameterizing / splitting up of tests

### CODE REVIEW

File: pandas/tests/reductions/test_reductions.py [hunk 1]
```diff
@@ -224,16 +228,17 @@ def test_minmax_timedelta64(self):
...
             assert idx.argmax() == 2
 
-        for op in ["min", "max"]:
-            # Return NaT
-            obj = TimedeltaIndex([])
-            assert pd.isna(getattr(obj, op)())
+    @pytest.mark.parametrize("op", ["min", "max"])
+    def test_minmax_timedelta_empty_or_na(self, op):
+        # Return NaT
+        obj = TimedeltaIndex([])
```
Comment (jreback) [inline pandas/tests/reductions/test_reductions.py]: explicity test for is NaT on the returns

File: pandas/tests/reductions/test_reductions.py [hunk 2]
```diff
@@ -66,60 +66,64 @@ def test_ops(self, opname, obj):
...
+        if klass is Series:
+            assert np.isnan(result)
+        else:
+            assert result == -1
+
+    @pytest.mark.parametrize("opname", ["max", "min"])
+    @pytest.mark.parametrize("dtype", ["M8[ns]", "datetime64[ns, UTC]"])
+    def test_nanops_empty_object(self, opname, index_or_series, dtype):
+        klass = index_or_series
+        arg_op = "arg" + opname if klass is Index else "idx" + opname
```
Comment (jreback) [inline pandas/tests/reductions/test_reductions.py]: why is this needed?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/reductions/test_reductions.py (hunk 2 above)
Comment (dsaxton) [reply pandas/tests/reductions/test_reductions.py]: I broke this into its own test since it allows for parametrizing over these dtypes without repeating all of `test_nanops` (which was / is pretty big), and this was conceptually a bit more specific than just "testing nanops"

Comment (jreback) [reply pandas/tests/reductions/test_reductions.py]: i get that, but why is arg_op specifically needed? what is its purpose

Comment (dsaxton) [reply pandas/tests/reductions/test_reductions.py]: Oh, I think the name of the method being tested changes if it's a Series or Index (idxmin or argmin) so it has to be created dynamically in the test?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dsaxton
